File: 747. Largest Number At Least Twice of Others/747.java
//747. Largest Number At Least Twice of Others solution in JavaScript
//https://leetcode.com/problems/largest-number-at-least-twice-of-others/description/

class Solution {
    public int dominantIndex(int[] nums) {
        int max_index = -1;
        int max_num = -1;
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > max_num) {
                max_num = nums[i];
                max_index = i;
            }
        }
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] * 2 > max_num && i != max_index) return -1;
        }
        
        return max_index;
    }
}

File: 747. Largest Number At Least Twice of Others/747.js
//747. Largest Number At Least Twice of Others solution in JavaScript
//https://leetcode.com/problems/largest-number-at-least-twice-of-others/description/

/**
 * @param {number[]} nums
 * @return {number}
 */
 var dominantIndex = function(nums) {
    let max_index = -1;
    let max_number = -1;
    
    for (let i = 0; i < nums.length; i++) {
        if (max_number < nums[i]) {
            max_number = nums[i];
            max_index = i;
        }
    }
    
    for (let i = 0; i < nums.length; i++) {
        if (i != max_index && 2* nums[i] > max_number) return -1;
    }
    
    return max_index;
};